mcts_new.py

```diff
@@ -0,0 +1,280 @@
+from __future__ import annotations
+
+import dataclasses
+import enum
+
+import einops
+import numpy as np
+import torch
+
+import skyjo as sj
+import skynet
+
+
+# MARK: Enums
+class MCTSSelectionMethod(enum.Enum):
+    UCB = "UCB"  # AlphaZero UCB
+    # Lots of different research ideas to explore here
+    # with other selection methods
+
+
+# MARK: Dataclasses
+@dataclasses.dataclass(slots=True)
+class DecisionStateNode:
+    state: sj.Skyjo
+    parent: AfterStateNode | None
+    prev_action: sj.SkyjoAction | None
```

Comment:
this is of course an absurd and functionally meaningless nit, but if we plan on publishing this, I would prefer names to be unabbreviated; for example, `previous_action` here and `visit_count` below. I just think it reads cleaner idk